packages/optimizer/lib/transformers/AmpBoilerplateErrorHandler.js

```diff
@@ -48,7 +59,11 @@ class AmpBoilerplateErrorHandler {
     }
 
     const errorHandler = createElement('script', {'amp-onerror': ''});
-    insertText(errorHandler, ERROR_HANDLER_TRANSFORMED);
+    if (params.esmModuleEnabled) {
```

Comment:
This could alternatively check if `html` has the `transformed` attribute:

```suggestion
    if (html.attribs.transformed !== undefined) {
```

cf. https://github.com/ampproject/amp-toolbox-php/blob/929220681fd80891dc6f110607269ec7127d5a0b/src/Optimizer/Transformer/AmpBoilerplateErrorHandler.php#L62-L64

And yet https://github.com/ampproject/amp-toolbox-php/pull/218.

Reply:
I think we need to decouple the ESM/onError logic from the transformed state. The onError handler needs to be changed depending on whether ESM is enabled or not. This can happen regardless of transformed state.

Reply:
> The onError handler needs to be changed depending on whether ESM is enabled or not.

I don't think this is the case. See https://github.com/ampproject/amp-toolbox-php/pull/218#issuecomment-854998921.

Reply:
But why shouldn't you be allowed to use ES6 modules in a source AMP document that does not use the Optimizer...?

Reply:
That is, ESM depends on `transformed` also being present. So the new `amp-onerror` _could_ be conditioned by whether ESM is present regardless of `transformed`, but since ESM depends on `transformed` we might as well just use it as the signal to use the new `amp-onerror`.

Reply:
> But why shouldn't you be allowed to use ES6 modules in a source AMP document that does not use the Optimizer...?

I think that should be allowed, but currently it is not. The use of `v0.mjs` depends on `transformed`: https://github.com/ampproject/amphtml/blob/65a6e3a3c0fd6837858a5600f0ee62a22ba4ee68/validator/validator-main.protoascii#L3150-L3174

Reply:
Weston is right, but there's still the theoretical option of a user explicitly disabling mjs for transformed AMP.